src/lib/comms/sol-lwm2m.c

```diff
@@ -629,6 +689,55 @@ fill_client_info(struct sol_lwm2m_client_info *cinfo,
 }
 
 static int
+fill_bootstrap_client_info(struct sol_lwm2m_bootstrap_client_info *bs_cinfo,
+    struct sol_coap_packet *req)
+{
+    uint16_t i, count;
+    bool has_name = false;
+    struct sol_str_slice query[5];
+    int r;
+
+    r = sol_coap_find_options(req, SOL_COAP_OPTION_URI_QUERY, query,
+        LWM2M_BOOTSTRAP_QUERY_PARAMS);
+    SOL_INT_CHECK(r, < 0, r);
+    count = r;
+
+    for (i = 0; i < count; i++) {
+        struct sol_str_slice key, value;
+        const char *sep;
+
+        SOL_DBG("Query:%.*s", SOL_STR_SLICE_PRINT(query[i]));
+        sep = memchr(query[i].data, '=', query[i].len);
+
+        if (!sep) {
+            SOL_WRN("Could not find the separator '=' at: %.*s",
+                SOL_STR_SLICE_PRINT(query[i]));
+            break;
+        }
+
+        key.data = query[i].data;
+        key.len = sep - query[i].data;
+        value.data = sep + 1;
+        value.len = query[i].len - key.len - 1;
+
+        if (sol_str_slice_str_eq(key, "ep")) {
+            //Required info
+            has_name = true;
+            bs_cinfo->name = sol_str_slice_to_str(value);
+            SOL_NULL_CHECK_GOTO(bs_cinfo->name, err_bs_cinfo_prop);
+        }
```

Comment:
btw this code above is wrong, because we're not setting the `r` value in case of a failure.
It should be
```c
bs_cinfo->name = sol_str_slice_to_str(value);
r = -ENOMEM;
SOL_NULL_CHECK_GOTO(bs_cinfo->name, err_bs_cinfo_prop);
....
```